frontend/packages/console-shared/src/hooks/useScrollShadows.tsx

```diff
@@ -43,3 +43,23 @@ export const useScrollShadows = (node: HTMLElement): Shadows => {
   }, [node, computeShadows]);
   return shadows;
 };
+
+type FormScrollShadowsContextType = { shadow?: Shadows };
+
+export const FormScrollShadowsContext = React.createContext<FormScrollShadowsContextType>({});
+
+type FormScrollShadowProviderProps = {
+  children: (callback: (node: HTMLElement) => void) => React.ReactNode;
+};
+
+export const FormScrollShadowProvider: React.FC<FormScrollShadowProviderProps> = ({ children }) => {
```

Comment:
Why do we need to keep the scroll position in a top level context here?

Reply:
Ref needs to be attached to the scrollable area and useScrollShadows need the ref that's why scrollPosition is in top level context